Maple2.Server.Game/PacketHandlers/PartyHandler.cs

```diff
@@ -0,0 +1,249 @@
+
+public class PartyHandler : PacketHandler<GameSession> {
+    public override RecvOp OpCode => RecvOp.Party;
+
+    private enum Command : byte {
+        Invite = 1,
+        InviteResponse = 2,
+        Leave = 3,
+        Kick = 4,
+        SetLeader = 17,
+        MatchPartyJoin = 23,
+        SummonParty = 29,
+        Unknown = 32,
+        PartySearch = 33,
+        CancelPartySearch = 34,
+        VoteKick = 45,
+        ReadyCheck = 46,
+        ReadyCheckResponse = 48
+    }
+
+    #region Autofac Autowired
+    // ReSharper disable MemberCanBePrivate.Global
+    public required WorldClient World { private get; init; }
+    // ReSharper restore All
+    #endregion
+
+    public override void Handle(GameSession session, IByteReader packet) {
+        var command = packet.Read<Command>();
+        switch (command) {
+            case Command.Invite:
+                HandleInvite(session, packet);
+                return;
+            case Command.InviteResponse:
+                HandleInviteResponse(session, packet);
+                return;
+            case Command.Leave:
+                HandleLeave(session, packet);
+                return;
+            case Command.Kick:
+                HandleKick(session, packet);
+                return;
+            case Command.SetLeader:
+                HandleSetLeader(session, packet);
+                return;
+            case Command.MatchPartyJoin:
+                HandleMatchPartyJoin(session, packet);
+                return;
+            case Command.SummonParty:
+                HandleSummonParty(session, packet);
+                return;
+            case Command.Unknown:
+                HandleUnknown(session, packet);
+                return;
+            case Command.PartySearch:
+                HandlePartySearch(session, packet);
+                return;
+            case Command.CancelPartySearch:
+                HandleCancelPartySearch(session, packet);
+                return;
+            case Command.VoteKick:
+                HandleVoteKick(session, packet);
+                return;
+            case Command.ReadyCheck:
+                HandleReadyCheck(session, packet);
+                return;
+            case Command.ReadyCheckResponse:
+                HandleReadyCheckResponse(session, packet);
+                return;
+        }
+
+    }
+
+    private void HandleInvite(GameSession session, IByteReader packet) {
+        if (session.Party.Party == null) {
+            // Create new party
+            PartyResponse response = World.Party(new PartyRequest {
+                RequestorId = session.CharacterId,
+                Create = new PartyRequest.Types.Create { },
+            });
+
+            var error = (PartyError) response.Error;
+            if (error != PartyError.none) {
+                session.Send(PartyPacket.Error(error));
+                return;
+            }
+
+            session.Party.SetParty(response.Party);
+        }
+
```

Comment:
we should validate that the player sending the invite is the leader of the party.